src/file.bash

```diff
@@ -489,7 +489,8 @@ assert_symlink_to() {
   fi
 
   local -r realsource=$( "${readlink_command[@]}" "$link" )
-  if [ ! "$realsource" = "$sourcefile"  ]; then
+  local -r realexpectedsource=$( "${readlink_command[@]}" "$sourcefile" )
```

Comment:
Please don't mix local and $(), it hides the exit code of the latter.

Reply:
I fixed the two occurrences in this function by replacing it with `local var; var=$(function)`, which avoids hiding the exit code but makes the variable mutable (non-read-only).
However this issue is present in many other places in the repo. I won't fix that in this PR to avoid making it a huge change, but I might submit a fix in a separate PR if you think it's a good idea.

Reply:
Cool, thanks. I've made the variable readonly. I'll give it a go at fixing some of these shellcheck issues and will make a new PR if I make some progress.